r2dbc-core/build.gradle

```diff
@@ -16,8 +16,7 @@ dependencies {
     api(mnValidation.micronaut.validation)
     api(libs.managed.r2dbc.spi)
 
-    api(libs.managed.reactor.netty.http)
-
+    api(platform(mnReactor.boms.reactor))
```

Comment:
@sdelamo This PR is now passing and it would be good if we can use it. Then we would not need this in micronaut-data https://github.com/micronaut-projects/micronaut-data/pull/3070/files#diff-697f70cdd88ba88fe77eebda60c7e143f6ad1286bca75017421e93ad84fb87dfR99
Before this change, we had netty-reactor dependencies coming via mariadb dependencies. And adding to micronaut-data just fixed issue we noticed in tests, but other users might need it too so it is better have api dependency in r2dbc than in data.